### DISCUSSION TITLE
Use Global parameter as dynamic value via in tokenisation (config file)

### DISCUSSION THREAD

**Discussion by ghost** [Azure-Player/azure.datafactory.tools#293] [Ideas]
Basically,

I have a (Azure) Terraform repo that creates a default databricks Cluter for each environment as its deployed (dev, qa, uat, prod). The associated ClusterID is then stored in ADF as a Global Parameter. 

When calling the "Publish-AdfV2FromJson", I "filter" global parameters from all environments (to maintain the ID created via Terraform), by filtering out the factory instances with "-factory.*". During this deployment, I would like to use the Global Parameter value when specifying a value for our Databricks Linked Service. Something like this:

"config-qa.csv"
--------------------
linkedService, lsDataBricksDeltaLakeLiab, parameters.prmClusterId.defaultValue, globalParameter.gpClusterId

Does this explanation make sense?

**Comment by NowinskiK** [comment]
Thanks for brining this idea. I'm pretty sure you know it's not working right now, but let me summarise why (maybe for others).
The process will not evaluate `globalParameter.gpClusterId`, where you need a real value to be replaced.
You have 2 options - choose one of them:
1) Declare parameter in LS and pass it from pipeline/dataset using value from Global Parameter.
2) Declare real value in config file replacing `globalParameter.gpClusterId`

Now, if you expect the process evaluating `globalParameter.gpClusterId` my concern is already about the step before:
How would you provide value for this Global parameter in the first place?
Because once you've got the value in the release pipeline - you can replace `globalParameter.gpClusterId` with placeholder which will be replaced by variable value.
